stackslib/src/chainstate/nakamoto/mod.rs

```diff
@@ -939,6 +939,17 @@ impl NakamotoChainState {
         Ok(())
     }
 
+    /// Get the height of the nakamoto staging block
+    pub fn get_nakamoto_block_height(
```

Comment:
Staging blocks aren't guaranteed to be processed.  I think you want to drop this method altogether, and use `NakamotoChainState::get_canonical_stacks_header()` to get the highest _processed_ block.

Reply:
EDIT: this might not be necessary at all; see below.